src/app/components/CollapsibleNavigation/index.tsx

```diff
@@ -0,0 +1,139 @@
+/** @jsx jsx */
+/* @jsxFrag React.Fragment */
+import { jsx } from '@emotion/react';
+import React, { useState, useRef } from 'react';
+import useHydrationDetection from '#app/hooks/useHydrationDetection';
+import styles from './index.styles';
+import { Close } from '../icons';
+import { CollapsibleNavigationSection } from './types';
+import Heading from '../Heading';
+
+type CollapsibleNavigationProps = {
+  navigationSections: CollapsibleNavigationSection[];
+};
+
+const CollapsibleNavigation = ({
+  navigationSections,
+}: CollapsibleNavigationProps) => {
+  const [openSection, setOpenSection] = useState<string | null>(null);
+  const isHydrated = useHydrationDetection();
+  const activeNavItemRef = useRef<HTMLAnchorElement | null>(null);
+
+  const handleNavClick = (
+    e: React.MouseEvent<HTMLAnchorElement>,
+    section: CollapsibleNavigationSection,
+  ) => {
+    if (section.href) {
+      return;
+    }
+
+    e.preventDefault();
+    const isActive = openSection === section.id;
+
+    if (isActive) {
+      setOpenSection(null);
+      activeNavItemRef.current = null;
+    } else {
+      activeNavItemRef.current = e.currentTarget;
+      setOpenSection(section.id);
+    }
+  };
+
+  const handleClose = (e: React.MouseEvent) => {
+    e.preventDefault();
+    setOpenSection(null);
+
+    if (activeNavItemRef.current) {
+      activeNavItemRef.current.focus();
+      activeNavItemRef.current = null;
+    }
+  };
+
+  return (
+    <nav role="navigation">
+      <ul role="list" css={styles.navList}>
+        {navigationSections.map(section => {
+          const isActive = Boolean(openSection === section.id);
+          const shouldShowSubNav = isHydrated ? isActive : true;
+          const isLink = section.href;
+
+          const navigationLinkId = `nav-${section.id}`;
+          const subNavigationTitleId = `subnav-title-${section.id}`;
+          const subNavigationId = section.id;
+
+          return (
+            <React.Fragment key={section.id}>
+              <li css={styles.navItem} role="listitem">
+                <a
+                  id={navigationLinkId}
+                  href={section.href || `#${section.id}`}
+                  onClick={e => handleNavClick(e, section)}
+                  className="focusIndicatorRemove"
+                  css={[styles.navLink, isActive && styles.navLinkActive]}
+                  role={isLink ? 'link' : 'button'}
+                  {...(isActive && { 'aria-current': 'true' })}
+                  {...(!isLink && { 'aria-expanded': isActive })}
+                  {...(!isLink && { 'aria-controls': subNavigationId })}
```

Comment:
Why don't we combine both of these?

```suggestion
                  {...(!isLink && { 'aria-expanded': isActive,  'aria-controls': subNavigationId })}
```

Reply:
Good idea! Updated